nx/lib/nx.ex

```diff
@@ -5244,6 +5244,39 @@ defmodule Nx do
     impl!(tensor).conjugate(%{tensor | type: Nx.Type.to_complex(tensor.type)}, tensor)
   end
 
+  @doc """
+  Calculates the complex phase angle of each element in the tensor.
+
+  If $z = a + bi = r e^\\theta$, $phase(z) = {\rm atan2}(b, a)$
+
+  ## Examples
+
+       iex> Nx.phase(Complex.new(1, 2))
+       #Nx.Tensor<
+         f32
+         1.1071487665176392
+       >
+
+       iex> Nx.phase(1)
+       #Nx.Tensor<
+         f32
+         0.0
+       >
+
+       iex> import Nx, only: [sigil_V: 2]
+       iex> Nx.phase(~V[1+2i -2+1i])
+       #Nx.Tensor<
+         f32[2]
+         [1.1071487665176392, 2.677945137023926]
+       >
+  """
+  @doc type: :element
+  def phase(tensor) do
+    tensor = to_tensor(tensor)
+
+    impl!(tensor).phase(%{tensor | type: Nx.Type.to_float(tensor.type)}, tensor)
```

Comment:
Looking at `Complex.phase`, could this not be implemented just as:

```suggestion
  def phase(tensor) do
    mask = less(tensor, 0)
    select(mask, Nx.Constants.pi(), atan2(imag(tensor), real(tensor))
  end
```

If so, it would save me from having to write something more verbose for EXLA :)

Reply:
Wouldn't this be atan2(imag, real) for the whole tensor?

x < 0 -> pi only makes sense for real numbers

Reply:
Yeah actually I see what you mean, so then it could just be `atan2(imag(tensor), real(tensor))`

And as an optional callback instead of with a BinaryBackend impl?

Reply:
Make sense! Optional imolementation could be the way to go

Reply:
> If so, it would save me from having to write something more verbose for EXLA :)

You would still need to use the same amount of verbosity for `Nx.real` and `Nx.imag`, as you need to iterate over the tensor. So, if you are doing for real and imag, why not for phase? Also, I think this would be more efficient than mapping with and anonymous function, especially if you're running natively.

However, I don't have any other issues with this approach and I am happy implement it this way.

Reply:
The advantage is that we'd only have to make mandatory callbacks (which force us to add grad impls) for real and imag.

You'll see, for instance, that Nx
LinAlg.determinant is a defn for the same reason this one should be, and invert has the perk of having a custom grad (because we didn't want to add the mandatory callback, but found a fast and easy impl for the grad)

Reply:
In short, it's not the worry of verbosity, but of the API surface that all backends need to implement